Source/Common/gdcmDirectory.cxx

```diff
@@ -73,8 +73,11 @@ unsigned int Directory::Explore(FilenameType const &name, bool recursive)
       }
     else
       {
-      Filenames.push_back(dirName+fileName);
-      nFiles++;
+      if (fileName[0] != '.') // discard "unix like" hidden files such as .git in submodules
```

Comment:
I do not understand this change ? why should I skip hidden files ? Technically we would need to do the same in the `WIN32` section below. Also the change is not backward compatible, so I'd rather have a new option `SkipHiddenDotFile()`

Reply:
To be consistent with the Unix code that does that (line 120).
You already did that  for directories but did not applied it on files:
https://github.com/malaterre/GDCM/commit/6a627a55f72f65a5d7d98188dc46b274e500e7a1
The TestScanner1 is failing due to a .git file that is added in Testing/Data and which is not expected by the test

Reply:
For reference line 120 points to commit 9e2791a9, 6a627a5 is about the hidden directory part.